sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -296,6 +296,20 @@ class Column(protected[sql] val expr: Expression) extends Logging {
   def eqNullSafe(other: Any): Column = this <=> other
 
   /**
+   * Between col1 and col2.
+   *
+   * @group java_expr_ops
+   */
+  def between(col1: String, col2: String): Column = between(Column(col1), Column(col2))
+
+  /**
+   * Between col1 and col2.
```

Comment:
"True if the current column is between the lower bound and upper bound, inclusive."